homeassistant/components/point/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for Minut Point."""
+from collections import OrderedDict
+
+import logging
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.core import callback
+
+
+CLIENT_ID = 'client_id'
+CLIENT_SECRET = 'client_secret'
+KEY_IMPORT_GROUPS = 'import_groups'
+
+AUTH_CALLBACK_PATH = '/api/minut'
+AUTH_CALLBACK_NAME = 'api:minut'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AuthError(Exception):
+    """Exception if authentication occurs."""
+
+    def __init__(self, code):
+        """Initialize exception."""
+        super().__init__()
+        self.code = code
+
+
+@config_entries.HANDLERS.register('point')
+class FlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize flow."""
+        self._client_id = None
+        self._client_secret = None
+        self._token = None
+        self._point_session = None
+        self._redirect_uri = None
+
+    async def token(self, code):
+        """Register token."""
+        token = self._point_session.get_access_token(code)
+        _LOGGER.debug("Got new token: %s", token)
+        if not self._point_session.is_authorized:
+            _LOGGER.error('Authentication Error')
+        else:
+            _LOGGER.info('Successfully authenticated Point')
+            self._token = token
+
+    async def async_step_user(self, user_input=None):
+        """Handle the authentication with a gateway."""
+        errors = {}
+
+        if user_input is not None:
+            self._client_id = user_input.get(
+                CLIENT_ID, self._client_id)
+            self._client_secret = user_input.get(
+                CLIENT_SECRET, self._client_secret)
+            return await self.async_step_auth()
+
+        fields = OrderedDict()
+        fields[vol.Required(CLIENT_ID, default='')] = str
+        fields[vol.Required(CLIENT_SECRET, default='')] = str
+        self._redirect_uri = self.hass.config.api.base_url + AUTH_CALLBACK_PATH
+
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema(fields),
+            errors=errors,
+            description_placeholders={'redirect_uri': self._redirect_uri}
+        )
+
+    async def async_step_auth(self, user_input=None):
+        """Create an entry for auth."""
+        if user_input is None or self._token is None:
```

Comment:
Looks like we need to change this check if we never expect any user input from the user. Perhaps:
```py
if self._token is None:
    authorization_url = await self.get_authorization_url()
    return self.async_show_form(
...
```

Reply:
We want to show the form and when the user have authenticated the keys in the first dialog the user will be redirected to home assistant and I'll set the token given the returned code.
*(This is the standard way oauth2 works.)*
Then I want the user to click submit and home assistant will go to `async_setup_auth` with `user_input` as `{} ` and we can create entities.

As I've been saying for the last week, it would be great if my `token` function could mimic a click on *setup* in the second dialog or if that dialog could be closed and we could call `create_entities` but this flow is good enough.
I'll be happy to update the flow when a reference implementation of a oauth2 redirect is in place..